source/agora/api/Validator.d

```diff
@@ -120,4 +121,22 @@ public interface API : agora.api.FullNode.API
     ***************************************************************************/
 
     public void receiveBlockSignature (ValidatorBlockSig block_sig);
+
+    /***************************************************************************
+
+        Params:
+            enroll_keys = set of enrollment keys
+            height = height
+
+        Returns:
+            The preimages for the specified enroll keys for the specified height.
+            If the requested node doesn't know the preimage for a specific
+            enroll key, then it will not be included in the result.
+
+        API:
+            GET /preimages_for_enroll_keys
+
+    ***************************************************************************/
+
```

Comment:
we could be more efficient here by only sharing the indices, but this will unlikely to cause any real problems